query-engine/connector-test-kit-rs/query-engine-tests/tests/new/regressions/prisma_15607.rs

```diff
@@ -0,0 +1,286 @@
+//! Testing deadlocks on SQL Server deadlocks and deadlock recovery.
+//! Due to certain queries hanging forever until the second
+//! transaction progresses, the test uses separate engines inside
+//! actors to allow test to continue even if one query is blocking.
+
+use std::future::Future;
+
+use indoc::indoc;
+use once_cell::sync::Lazy;
+use query_engine_tests::{
+    query_core::TxId, render_test_datamodel, setup_metrics, setup_project, test_tracing_subscriber, ConnectorTag,
+    QueryResult, Runner, TestConfig, TestError, TestResult, TryFrom, WithSubscriber, ENV_LOG_LEVEL,
+};
+use tokio::sync::mpsc;
+
+static CONFIG: Lazy<TestConfig> = Lazy::new(|| TestConfig::load().unwrap());
+
+const SCHEMA: &str = indoc! {r#"
+    model Country {
+      id         Int    @id
+      name       String
+      cities     City[]
+    }
+
+    model City {
+      id         Int    @id
+      country_id Int
+      name       String
+      country    Country @relation(fields: [country_id], references: [id])
+    }
+"#};
+
+struct Actor {
+    query_sender: mpsc::Sender<Message>,
+    response_receiver: mpsc::Receiver<Response>,
+}
+
+#[derive(Debug, Clone)]
+enum Message {
+    Query(&'static str),
+    BeginTransaction,
+    RollbackTransaction(TxId),
+    SetActiveTx(TxId),
+}
+
+#[derive(Debug)]
+enum Response {
+    Query(TestResult<QueryResult>),
+    Tx(TestResult<TxId>),
+    Rollback(Result<(), user_facing_errors::Error>),
+}
```

Comment:
one alternative pattern I like is sending closures through the channel. But this is also good, for code that has to be maintained :+1:

Reply:
More like erlang style...